Move integration-cli daemon package to internal/test…

vdemeester opened the pull request:
… and do not use the `docker` cli in it. One of the reason of this
move is to not make `integration` package using legacy
`integration-cli` package.

Next move will be to support swarm within this package *and* provide
some helper function using the api (compared to the one using cli in
`integration-cli/daemon` package).

:lion: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented on integration-cli/check_test.go:
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/docker/docker/integration-cli/environment"
 	"github.com/docker/docker/integration-cli/fixtures/plugin"
 	"github.com/docker/docker/integration-cli/registry"
+	idaemon "github.com/docker/docker/internal/test/daemon"
```
How come we are using an alias for this?
If we must (which would seem a bit off since I don't think we should be importing the real daemon package here), then maybe `testdaemon` is better?

vdemeester replied:
yeah `testdaemon` is probably better. We have to use an alias for this right now as this package still refers both. It's in the "Next move"/follow-up PR that I'll try to clean that and migrate all "cli" helpers in `integration-cli/daemon` to api helpers that `internal/test/daemon` package :angel:.
I just didn't want to make a *huge* PR that would be really hard to review...

cpuguy83 reviewed (approved):
LGTM

Just a minor nit.

vdemeester committed:
Move integration-cli daemon package to internal/test…

… and do not use the `docker` cli in it. One of the reason of this
move is to not make `integration` package using legacy
`integration-cli` package.

Next move will be to support swarm within this package *and* provide
some helper function using the api (compared to the one using cli in
`integration-cli/daemon` package).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

yongtang reviewed (approved):
LGTM, Thanks!

thaJeztah commented:
This one will conflict with https://github.com/moby/moby/pull/36722, so lets get that one merged first

thaJeztah reviewed (approved):
LGTM

thaJeztah commented:
Chatted with @vdemeester and the other way round is easier, so let me merge this 👍
